Controller/CRUDController.php

```diff
@@ -265,6 +265,9 @@ public function editAction($id = null)
 
             // persist if the form was valid and if in preview mode the preview was approved
             if ($isFormValid && (!$this->isInPreviewMode() || $this->isPreviewApproved())) {
+                $object = $form->getData();
+                $this->admin->setSubject($object);
```

Comment:
If new object is not created in a form as I provided in sample DataMapper in corresponding issue then there is no need to get data from form because instance remains the same. But otherwise `$object` in the method and data in the form are different. That's why there is a need to get data.